keps/420-partial-admission/README.md

```diff
@@ -0,0 +1,139 @@
+
+### Non-Goals
+
+Since this is an opt-in feature, the parent job should accept the partial admission parameters provided by Kueue.
+
+Kueue will not take any measure to ensure that the parent job respects the assigned quota.
+
+## Proposal
+
+Change the way the flavor assigner work to support decrementing the pods count in order to find a better fit for the current workload.
+In case a partial fit is chosen, the jobframework reconciler should provide the admitted pod counts to the parent job before unsuspending it, in a similar fashion as the node selectors are provided. In case the job gets suspended, the original pod counts should be restored in order to allow a potential future admission with its original pod counts.
+
+### User Stories
+
+Kueue issue [420](https://github.com/kubernetes-sigs/kueue/issues/420) provides details on the initial feature details and its applicability for `batch/Job`.
+
+### Risks and Mitigations
+
+TBD: It should be decided if the partial admission should be taken into account before attempting preemption. Since preemption has its own penalties (time to evict the active workloads, potential waste of the resources used by the preempted workloads) I vote to take it into account before.
+
+## Design Details
+
+### Workload API
+
+```go
+type PodSet struct {
+    // .......
+
+    // count is the number of pods for the spec.
+    // +kubebuilder:validation:Minimum=1
+    Count int32 `json:"count"`
+
+    // minimumCount the minimum caount acceoted for this
+    // PodSet in case of partial admission.
+    // If not provided, partial admission is not accepted
+    // for this PodSet
+    // +optional
+    MinimumCount *int32 `json:"minimumCount"`
+}
+
+```
+
+### Scheduler / Flavorassignment
+
+I case the current workload dose not in the current available quota and any of its PodSets allow partial admission and the cont did not reach the `MinimumCount`, decrease the cont in one PodSet and try again. The accepted number of pods in each PodSet are published in `workload.Status.Admission.PodSetAssignments[*].ResourceUsage.Count`
+
+### Jobframework
+
+```diff
+type GenericJob interface {
+    // ...
+
+	
+-    // RunWithNodeAffinity will inject the node affinity extracting from workload to job and unsuspend the job.
++    // RunWithPodSetsInfo will inject the node affinity and podsSet counts extracted from workload to the job and unsuspend the job.
+-    RunWithNodeAffinity(nodeSelectors []PodSetNodeSelector)
++    RunWithPodSetsInfo(nodeSelectors []PodSetNodeSelector, podSetCounts []int32)
+-    // RestoreNodeAffinity will restore the original node affinity of job.
++    // RestorePodSetsInfo will restore the original node affinity of job.
+-    RestoreNodeAffinity(nodeSelectors []PodSetNodeSelector)
++    RestorePodSetsInfo(nodeSelectors []PodSetNodeSelector, podSetCounts []int32)
+
+    // ...
+}
+
+```
+
+### batch/Job controller
```

Comment:
What happens if clusterQueue can admit a minimal workload (partial admission) by preempting the running workload?

Does the scheduler preempt running workloads following the `PreemptionPolicy` similar to the case of not using partial admission?